diff --git a/drivers/net/mlx5/mlx5_flow_hw.c b/drivers/net/mlx5/mlx5_flow_hw.c
index 6dc16f8..3f21cd8 100644
--- a/drivers/net/mlx5/mlx5_flow_hw.c
+++ b/drivers/net/mlx5/mlx5_flow_hw.c
@@ -8893,6 +8893,8 @@ pattern_template_validate(struct rte_eth_dev *dev,
 			  uint32_t pt_num,
 			  struct rte_flow_error *error)
 {
+	return 0;
+
 	struct mlx5_flow_template_table_cfg tbl_cfg = {
 		.attr = {
 			.nb_flows = 64,
