diff --color -ruN dpdk-23.11/lib/log/log.c dpdk-23.11-patch/lib/log/log.c
--- dpdk-23.11/lib/log/log.c	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/log/log.c	2024-03-04 15:01:39.318300347 +0000
@@ -85,6 +88,13 @@
 	return 0;
 }
 
+/* Change the print function that will be used by the logging system. */
+void
+rte_log_set_print_func(int (*func)(FILE *stream, const char *format, va_list ap))
+{
+	rte_logs.print_func = func;
+}
+
 FILE *
 rte_log_get_stream(void)
 {
diff --color -ruN dpdk-23.11/lib/log/rte_log.h dpdk-23.11-patch/lib/log/rte_log.h
--- dpdk-23.11/lib/log/rte_log.h	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/log/rte_log.h	2024-03-04 15:03:08.442486094 +0000
@@ -89,6 +89,18 @@
 int rte_openlog_stream(FILE *f);
 
 /**
+ * Change the print function that will be used by the logging system.
+ *
+ * This can be done at any time. The func argument represents
+ * the vfprintf-like function to be used to print the logs.
+ * Without calling this function, the default (vfprintf) is used.
+ *
+ * @param func
+ *   Pointer to the print function.
+ */
+void rte_log_set_print_func(int (*func)(FILE *stream, const char *format, va_list ap));
+
+/**
  * Retrieve the stream used by the logging system (see rte_openlog_stream()
  * to change it).
  *
diff --color -ruN dpdk-23.11/lib/log/version.map dpdk-23.11-patch/lib/log/version.map
--- dpdk-23.11/lib/log/version.map	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/log/version.map	2024-03-04 15:03:23.118516680 +0000
@@ -17,6 +17,7 @@
 	rte_log_set_level_pattern;
 	rte_log_set_level_regexp;
 	rte_openlog_stream;
+	rte_log_set_print_func;
 	rte_vlog;

 	local: *;
