diff --color -ruN dpdk-23.11/lib/ethdev/rte_ethdev.c dpdk-23.11-patch/lib/ethdev/rte_ethdev.c
--- dpdk-23.11/lib/ethdev/rte_ethdev.c	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/ethdev/rte_ethdev.c	2024-03-04 12:28:35.492608466 +0000
@@ -428,7 +428,7 @@
 	return 1;
 }

-uint64_t
+uint16_t
 rte_eth_find_next_owned_by(uint16_t port_id, const uint64_t owner_id)
 {
 	port_id = rte_eth_find_next(port_id);
diff --color -ruN dpdk-23.11/lib/ethdev/rte_ethdev.h dpdk-23.11-patch/lib/ethdev/rte_ethdev.h
--- dpdk-23.11/lib/ethdev/rte_ethdev.h	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/ethdev/rte_ethdev.h	2024-03-04 12:28:28.084570580 +0000
@@ -2126,7 +2126,7 @@
  * @return
  *   Next valid port ID owned by owner_id, RTE_MAX_ETHPORTS if there is none.
  */
-uint64_t rte_eth_find_next_owned_by(uint16_t port_id,
+uint16_t rte_eth_find_next_owned_by(uint16_t port_id,
 		const uint64_t owner_id);

 /**
