diff --git a/drivers/net/mlx5/mlx5_flow_dv.c b/drivers/net/mlx5/mlx5_flow_dv.c
index 7b9e501..bc324ff 100644
--- a/drivers/net/mlx5/mlx5_flow_dv.c
+++ b/drivers/net/mlx5/mlx5_flow_dv.c
@@ -7927,13 +7927,7 @@ flow_dv_validate(struct rte_eth_dev *dev, const struct rte_flow_attr *attr,
 				mlx5_flow_tunnel_ip_check(items, next_protocol,
 							  item_flags,
 							  &l3_tunnel_flag);
-			/*
-			 * explicitly allow inner IPIP match
-			 */
-			if (l3_tunnel_detection == l3_tunnel_outer) {
-				item_flags |= l3_tunnel_flag;
-				tunnel = 1;
-			}
+
 			ret = mlx5_flow_dv_validate_item_ipv4(dev, items,
 							      item_flags,
 							      last_item,
@@ -7954,13 +7948,7 @@ flow_dv_validate(struct rte_eth_dev *dev, const struct rte_flow_attr *attr,
 				mlx5_flow_tunnel_ip_check(items, next_protocol,
 							  item_flags,
 							  &l3_tunnel_flag);
-			/*
-			 * explicitly allow inner IPIP match
-			 */
-			if (l3_tunnel_detection == l3_tunnel_outer) {
-				item_flags |= l3_tunnel_flag;
-				tunnel = 1;
-			}
+
 			ret = mlx5_flow_validate_item_ipv6(dev, items,
 							   item_flags,
 							   last_item,
