diff --color -ruN dpdk-23.11/lib/log/log.c dpdk-23.11-patch/lib/log/log.c
--- dpdk-23.11/lib/log/log.c	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/log/log.c	2024-03-04 15:01:39.318300347 +0000
@@ -31,11 +31,14 @@
 	uint32_t type;  /**< Bitfield with enabled logs. */
 	uint32_t level; /**< Log level. */
 	FILE *file;     /**< Output file set by rte_openlog_stream, or NULL. */
+	/** Print function set by rte_log_set_print_func, or vfprintf. */
+	int (*func)(FILE *stream, const char *format, va_list ap);
 	size_t dynamic_types_len;
 	struct rte_log_dynamic_type *dynamic_types;
 } rte_logs = {
 	.type = UINT32_MAX,
 	.level = RTE_LOG_DEBUG,
+	.func = vfprintf,
 };
 
 struct rte_eal_opt_loglevel {
@@ -80,6 +83,13 @@
 	return 0;
 }
 
+/* Change the print function that will be used by the logging system. */
+void
+rte_log_set_print_func(int (*func)(FILE *stream, const char *format, va_list ap))
+{
+	rte_logs.func = func;
+}
+
 FILE *
 rte_log_get_stream(void)
 {
@@ -501,7 +511,7 @@
 	RTE_PER_LCORE(log_cur_msg).loglevel = level;
 	RTE_PER_LCORE(log_cur_msg).logtype = logtype;
 
-	ret = vfprintf(f, format, ap);
+	ret = rte_logs.func(f, format, ap);
 	fflush(f);
 	return ret;
 }
diff --color -ruN dpdk-23.11/lib/log/rte_log.h dpdk-23.11-patch/lib/log/rte_log.h
--- dpdk-23.11/lib/log/rte_log.h	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/log/rte_log.h	2024-03-04 15:03:08.442486094 +0000
@@ -88,6 +88,18 @@
 int rte_openlog_stream(FILE *f);
 
 /**
+ * Change the print function that will be used by the logging system.
+ *
+ * This can be done at any time. The func argument represents
+ * the vfprintf-like function to be used to print the logs.
+ * Without calling this function, the default (vfprintf) is used.
+ *
+ * @param func
+ *   Pointer to the print function.
+ */
+void rte_log_set_print_func(int (*func)(FILE *stream, const char *format, va_list ap));
+
+/**
  * Retrieve the stream used by the logging system (see rte_openlog_stream()
  * to change it).
  *
diff --color -ruN dpdk-23.11/lib/log/version.map dpdk-23.11-patch/lib/log/version.map
--- dpdk-23.11/lib/log/version.map	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/log/version.map	2024-03-04 15:03:23.118516680 +0000
@@ -17,6 +17,7 @@
 	rte_log_set_level_pattern;
 	rte_log_set_level_regexp;
 	rte_openlog_stream;
+	rte_log_set_print_func;
 	rte_vlog;

 	local: *;
