diff --color -ruN dpdk-23.11/lib/telemetry/telemetry_data.c dpdk-23.11-patch/lib/telemetry/telemetry_data.c
--- dpdk-23.11/lib/telemetry/telemetry_data.c	2023-11-28 14:35:41.000000000 +0000
+++ dpdk-23.11-patch/lib/telemetry/telemetry_data.c	2024-03-04 13:35:15.585285834 +0000
@@ -141,7 +141,7 @@
 valid_name(const char *name)
 {
 	/* non-alphanumeric characters allowed in names */
-	static const char allowed[128] = { ['_'] = 1, ['/'] = 1 };
+	static const char allowed[128] = { ['_'] = 1, ['/'] = 1, ['.'] = 1, [':'] = 1, ['-'] = 1 };

 	for (; *name != '\0'; name++) {
 		if (isalnum(*name))
